sdk/python/src/dagger/cli.py

```diff
@@ -26,15 +27,13 @@ def generate(
     """
     # not using `dagger.Connection` because codegen is
     # generating the client that it returns
-    connector = get_connector(Config())
-    transport = connector.make_sync_transport()
+    cfg = Config()
 
-    with connector.make_graphql_client(transport) as session:
-        if session.client.schema is None:
-            raise typer.BadParameter(
-                "Schema not initialized. Make sure the dagger engine is running."
-            )
-        code = codegen.generate(session.client.schema)
+    if cfg.host.scheme == "docker-image":
+        with Engine(cfg) as engine:
+            code = generate_code(engine.cfg)
+    else:
+        code = generate_code(engine.cfg)
```

Comment:
@helderco I'm guessing this got messed up while I was rebasing, but I'm not sure what the original intention was 😔 Once this is taken care this should be good to merge.

Reply:
Yeah, the second one should be just `generate_code(cfg)` :)